docs/content/dagster-plus/managing-deployments/branch-deployments/using-branch-deployments-with-github.mdx

```diff
@@ -30,87 +33,108 @@ To complete the steps in this guide, you'll need:
 
 ## Step 1: Generate a Dagster+ agent token
 
+<Note>
+  <strong>If using Serverless deployment</strong>, skip this step.
+</Note>
+
 <GenerateAgentToken />
 
 ---
 
 ## Step 2: Create and configure an agent
 
 <Note>
-  <strong>
-    If using{" "}
-    <a href="/dagster-plus/deployment/serverless">Serverless deployment</a>
-  </strong>
-  , this step isn't required.
+  <strong>If using Serverless deployment</strong>, skip this step.
 </Note>
 
 <BDCreateConfigureAgent />
 
 ---
 
-## Step 3: Add GitHub workflow files to your repository
+## Step 3: Set up the GitHub Action files
 
 <Note>
-  If you used the GitHub app to add the repository as a code location, this step
-  isn't required.{" "}
+  <strong>If you used the Dagster+ GitHub app to import your project</strong>,
+  this step was already done for you.{" "}
   <a href="#step-4-configure-the-github-repository">Skip to the next step</a>.
 </Note>
 
-In this step, you'll add the required GitHub workflow files to your GitHub repository. The files can be found in our [Hybrid quickstart repository](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart) or [Serverless quickstart repository](https://github.com/dagster-io/dagster-cloud-serverless-quickstart), depending on the agent you are using.
+In this step, you'll add the required GitHub Action files to your GitHub repository. The files can be found in the [Hybrid quickstart repository](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart) or [Serverless quickstart repository](https://github.com/dagster-io/dagster-cloud-serverless-quickstart), depending on the deployment type you're using.
 
-Copy the following files to your repository (the linked files are shown in the Hybrid repo, there are equivalents in the Serverless repo).
+<TabGroup>
+<TabItem name="Hybrid">
 
-- [`dagster_cloud.yaml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/dagster_cloud.yaml)
-- [`.github/workflows/dagster-cloud-deploy.yml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/.github/workflows/dagster-cloud-deploy.yml)
+### Hybrid
 
-In the next step, you'll modify these files to work with your Dagster+ setup.
+1. Add the following files to the repository:
 
----
+   - [`dagster_cloud.yaml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/dagster_cloud.yaml)
+   - [`.github/workflows/dagster-cloud-deploy.yml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/.github/workflows/dagster-cloud-deploy.yml)
 
-## Step 4: Configure the GitHub repository
+2. In the [`dagster_cloud.yaml` file](/dagster-plus/managing-deployments/dagster-cloud-yaml), replace `build.registry` with the registry used by the [agent you created in Step 2](#step-2-create-and-configure-an-agent). For example:
```

Comment:
Probably missing something very basic here, but I'm a bit confused by adding the registry here. If this file is getting committed, wouldnt the agent specified in `build.registry` also be used when deployed to prod? Even through we created the agent with the intent for it to be used for branch deployments

Since this section has existed since this guide was first written, this question isn't blocking in any way, more just for my own knowledge

Reply:
prod and branch deployments use different dagster_cloud.yaml files, so it has to be specfified in both. I could imagine users using a different docker registry for their branch deployment images vs. their prod ones

Reply:
lol now I feel like I'm missing something. This is how I'm now interpreting this - is this correct?

- There's a `dagster_cloud.yaml`, set via the UI, that imports the project and allows the `prod` deployment to work
- There's another `dagster_cloud.yaml` in the GH repository that sets the agent registry and allows Branch Deps to work